packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -192,98 +289,226 @@ class _LogMessage {
     String prefix = '${millis.toString().padLeft(4)} ms • ';
     String indent = ''.padLeft(prefix.length);
     if (millis >= 100)
-      prefix = terminal.writeBold(prefix.substring(0, prefix.length - 3)) + ' • ';
+      prefix = terminal.toBold(prefix.substring(0, prefix.length - 3)) + ' • ';
     String indentMessage = message.replaceAll('\n', '\n$indent');
 
     if (type == _LogType.error) {
-      stderr.writeln(prefix + terminal.writeBold(indentMessage));
+      stderr.writeln(prefix + terminal.toBold(indentMessage));
       if (stackTrace != null)
         stderr.writeln(indent + stackTrace.toString().replaceAll('\n', '\n$indent'));
     } else if (type == _LogType.status) {
-      print(prefix + terminal.writeBold(indentMessage));
+      print(prefix + terminal.toBold(indentMessage));
     } else {
       print(prefix + indentMessage);
     }
   }
 }
 
+String _tputGetSequence(String capName, {String orElse}) {
+  if (AnsiTerminal._testMode) {
+    return '[$capName]';
+  }
+  if (Platform.isWindows) {
+    return orElse;
+  }
+  ProcessResult result =
+      Process.runSync('tput',  <String>['$capName'], stdoutEncoding:UTF8);
```

Comment:
I meant per-invocation, yeah. I agree that caching to a file wouldn't work (and at the point where you're hitting disk, you're probably not saving anything over just running `tput` anyway).